api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
 from marshmallow import ValidationError
 
 from . import groups
-from .models import GroupSchema, Group, Attendance, Member, MemberSchema, Meeting, MeetingSchema, AttendanceSchema
+from .models import Group, GroupSchema, Attendance, Member, MemberSchema, Meeting, MeetingSchema, AttendanceSchema, Manager, ManagerSchema, GroupType, GroupTypeSchema, ManagerType, ManagerTypeSchema
 from .. import db
 from ..images.models import Image, ImageGroup
-from ..people.models import Role, Manager, Person
+from ..people.models import Role, Person
+from src.shared.helpers import modify_entity, get_all_queried_entities
 
-# ---- Group
+from src.shared.models import QueryArgumentError
 
-group_schema = GroupSchema()
+from sqlalchemy.exc import IntegrityError, DBAPIError
 
+# ---- Group Type
 
-def group_dump(group):
-    group.manager_info = group.manager
-    group.manager_info.person = group.manager.person
-    group.member_list = group.members
-    return jsonify(group_schema.dump(group))
+group_type_schema = GroupTypeSchema()
 
+@groups.route('/group-types', methods=['POST'])
+def create_group_type():
+    try:
+        valid_group_type = group_type_schema.load(request.json)
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
-@groups.route('/groups', methods=['POST'])
+    new_group_type = GroupType(**valid_group_type)
+    db.session.add(new_group_type)
+    db.session.commit()
+    return jsonify(group_type_schema.dump(new_group_type)), 201
+
+@groups.route('/group-types/<int:group_type_id>', methods=['GET'])
+def read_one_group_type(group_type_id):
+    group_type = db.session.query(GroupType).filter_by(id=group_type_id).first()
+    if not group_type:
+        return jsonify(f"GroupType with id #{group_type_id} does not exist."), 404
```

Comment:
This construct of `jsonify(...some text ...)` appears fairly often in the code.  It seems kind of weird, in that a string is _already_ JSON. I wonder if this pattern has seeped into the code from previous developers using copy-and-paste? Take a look at how the UI deals with this kind of response. Is there a way to make it more general?

Reply:
Yeah. I believe this kind of construct has been in the code base for a pretty long time. I took that construct from the `events` module. And after a quick glance at the UI code, I believe in many places, the UI simply leaves error messages unhandled, so it won't make much of a difference what kind of object we return from the API upon error. But anyhow, do you have any recommendation regarding how we should change that construct? Maybe just return a string instead?

Reply:
This is going to require more thought.